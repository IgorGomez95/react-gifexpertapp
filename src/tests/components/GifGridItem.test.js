import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('pruebas para el componente GifGridItem ', () => {
    const title = 'test title';
    const url = 'http://localhost/test.jpg';
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
    test('debe de mostrar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de tener un pÃ¡rrafo con el title ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('debe de tener una imagen igual al url y alt de las props ', () => {
        const img = wrapper.find('img');
        // console.log( img.props() ); //Devuelve un objeto con las propiedades del elemento
        // console.log( img.prop('src') );//Devuelve solo la propiedad que se busca
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('debe de tener la clase animate__zoomIn ', () => {
        const div = wrapper.find('div');
        expect( div.hasClass('animate__zoomIn') ).toBe(true);
    })
    
    
    
});
