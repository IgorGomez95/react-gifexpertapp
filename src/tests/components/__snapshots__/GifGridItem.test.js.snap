// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas para el componente GifGridItem  debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="test title"
    src="http://localhost/test.jpg"
  />
  <p>
    test title
  </p>
</div>
`;
